libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -70,23 +70,34 @@ WasmDialect::WasmDialect()
 	addFunction("i64.extend_i32_u", {i32}, {i64});
 
 	addFunction("i32.store", {i32, i32}, {}, false);
-	m_functions["i32.store"_yulstring].sideEffects.invalidatesStorage = false;
+	m_functions["i32.store"_yulstring].sideEffects.storage = SideEffects::Pure;
+	m_functions["i32.store"_yulstring].sideEffects.memory = SideEffects::Read;
```

Comment:
Shouldn't this be `write`?

Reply:
I don't really get wasm yet. But to be equivalent to the old code, it should be `write`, so I'll fix it. I'll set all loads as read and stores and write.

Reply:
`store` is `mstore` and `load` is `mload` - wasm does not have a concept of storage - storage is implemented via an imported external function